### DISCUSSION TITLE
Can SerializeAs be a function?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#3440] [Help] [answered]
Can we call `serializeAs` with a function instead of a string? I would like to remove the fields conditionally depending on the model instance attributes.

```ts
export default class User extends BaseModel {
  @column({ serializeAs: () => ... })
  public password: string
}
```

https://docs.adonisjs.com/guides/models/serializing-models#hiding-properties

**Comment by thetutlage** [comment, marked as answer]
You can define your own `serializeAttributes` function on the model to customize the serialization behavior.

```ts
export default class User extends BaseModel {
  public serializeAttributes() {
    // your serialization logic
  }
}
```

Here is the default implementation https://github.com/adonisjs/lucid/blob/develop/src/Orm/BaseModel/index.ts#L1797
